crates/bonsaidb-client/src/client/remote_database.rs

```diff
@@ -156,6 +156,40 @@ impl<A: CustomApi> Connection for RemoteDatabase<A> {
         }
     }
 
+    async fn list_headers<C, R, PrimaryKey>(
+        &self,
+        ids: R,
+        order: Sort,
+        limit: Option<u32>,
+    ) -> Result<Vec<Header>, bonsaidb_core::Error>
+    where
+        C: Collection,
+        R: Into<Range<PrimaryKey>> + Send,
+        PrimaryKey: Into<AnyDocumentId<C::PrimaryKey>> + Send,
+    {
+        match self
+            .client
+            .send_request(Request::Database {
+                database: self.name.to_string(),
+                request: DatabaseRequest::List {
```

Comment:
```suggestion
                request: DatabaseRequest::ListHeaders {
```

Reply:
Now that clippy isn't causing the tests to not run, the client/server code isn't passing unit tests (`cargo test --package bonsaidb --test core-suite --all-features`). This change should fix it.